### PR TITLE
Updated the godoc on the CRD API types to follow conventions

### LINKED ISSUE #81519: clean up CRD type godoc
follow-up from https://github.com/kubernetes/kubernetes/pull/79604#discussion_r314407261

Update the godoc on the CRD API types to follow conventions by matching the serialized field name

/sig api-machinery
/kind cleanup
/assign
/priority important-soon
/area custom-resources

### PR DESCRIPTION
**Pull request by Zyqsempai**
/kind cleanup

Updated the godoc on the CRD API types to follow conventions

Fixes #81519

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go [hunk 1]
```diff
@@ -34,13 +34,13 @@ const (
 
 // CustomResourceDefinitionSpec describes how a user wants their resource to appear
 type CustomResourceDefinitionSpec struct {
-	// Group is the group this resource belongs in
+	// `group` is the group this resource belongs in
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go]: these should be unquoted and match the json tag

**Comment by liggitt** [comment]
thanks for the PR. please also update the types in the v1beta1 types.go file.

you will need to regenerate documentation with `make update`

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go (hunk 1 above)
Comment (Zyqsempai) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go]: Come again, all the names should be unquoted?

Comment (liggitt) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go]: yes, the godoc should start with the json tag, unquoted

**Comment by Zyqsempai** [comment]
@liggitt PTAL

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go [hunk 2]
```diff
@@ -338,23 +338,23 @@ type CustomResourceDefinitionList struct {
...
 
-	// Items individual CustomResourceDefinitions
+	// items individual CustomResourceDefinitions
 	Items []CustomResourceDefinition `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
 // CustomResourceValidation is a list of validation methods for CustomResources.
 type CustomResourceValidation struct {
-	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
+	// `openAPIV3Schema` is the OpenAPI v3 schema to be validated against.
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types.go]: drop quotes

File: build/root/BUILD.root [hunk 3]
```diff
@@ -89,6 +89,7 @@ filegroup(
         "//cmd:all-srcs",
         "//docs:all-srcs",
         "//hack:all-srcs",
+        "//klog:all-srcs",
```
Comment (liggitt) [inline build/root/BUILD.root]: this doesn't seem correct... probably need to drop this change

**Comment by liggitt** [comment]
two nits, lgtm otherwise, thanks!

**Comment by Zyqsempai** [comment]
@liggitt Done

**Comment by liggitt** [comment]
the verify job indicates the static openapi spec needs updating:

> /home/prow/go/src/k8s.io/kubernetes/api/openapi-spec is out of date. Please run hack/update-openapi-spec.sh

**Pull request closed without merging**